lib/util/cache/repository/impl/s3.ts

```diff
@@ -0,0 +1,70 @@
+import { Readable } from 'stream';
+import {
+  GetObjectCommand,
+  GetObjectCommandInput,
+  PutObjectCommand,
+  PutObjectCommandInput,
+} from '@aws-sdk/client-s3';
+import { logger } from '../../../../logger';
+import { getS3Client, parseS3Url } from '../../../s3';
+import { streamToString } from '../../../streams';
+import type { RepoCacheRecord } from '../types';
+import { RepoCacheBase } from './base';
+
+export class RepoCacheS3 extends RepoCacheBase {
+  private readonly s3Client;
+  private readonly bucket;
+
+  constructor(repository: string, url: string) {
+    super(repository);
+    this.bucket = parseS3Url(url)?.Bucket;
+    this.s3Client = getS3Client();
+  }
+
+  async read(): Promise<string | null> {
+    const cacheFileName = this.getCacheFileName();
+    const s3Params: GetObjectCommandInput = {
+      Bucket: this.bucket,
+      Key: cacheFileName,
+    };
+    try {
+      const { Body: res } = await this.s3Client.send(
```

Comment:
any reason why using `send` instead of simpler `getObject` function?

https://github.com/renovatebot/renovate/blob/a5b9ad096bc8ec5706849218de332deb4b983b2f/lib/modules/datasource/maven/util.ts#L116

Reply:
getObject is sdk v2, we are using v3. this is how amazon docs does it
and getObject is harder to test. wasnt able really

![image](https://user-images.githubusercontent.com/97394622/186615243-b42694e3-72a2-4120-a52a-ebe8fc2de432.png)
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/index.html

p.s. try finding docs related section on getObject in v3, i couldnt 😓

Reply:
ok, so we should update the other maven code to follow sdk v3, i think that's v2 compatibillity code there